src/concepts/components/properties.md

```diff
@@ -1,26 +1,551 @@
+    type Properties = Props;
+
+    fn create(props: Self::Properties, link: ComponentLink<Self>) -> Self {
+        Self {
+            link,
+            props: props.clone(),
+            name: props.name.into(),
+            show_message: false,
+        }
+    }
+
+    fn update(&mut self, msg: Self::Message) -> ShouldRender {
+        match msg {
+            Msg::Click() => self.show_message = true,
+        }
+        true
+    }
+
+    fn change(&mut self, props: Self::Properties) -> ShouldRender {
+        if self.props != props {
+            self.props = props;
+            true
+        } else {
+            false
+        }
+    }
+
+    fn view(&self) -> Html {
+        if !self.show_message {
+            html! {
+                <>
+                    <button onclick=self.link.callback( |_| Msg::Click() )>{"Click here!"}</button>
+                </>
+            }
+        } else {
+            html! {
+                <>
+                    <h1>{format!("Hello {}",self.name)}</h1>
+                </>
+            }
+        }
+    }
+}
+
+```
+
+In order to use this component you have to:
+
+```html
+// ...
+    <div class="full-height">
+        {"In this example we pass the name as parameter of the Yew component."}
+        <UseOfPropertyComponent name="Clark"/>
+    </div>
+// ...
+
+```
+
+## Define the properties structure
+
+```rust
+# use yew::prelude::*;
+#
+# pub struct UseOfPropertyComponent {
+#     link: ComponentLink<Self>,
+#     props: Props,
+#     name: String,
+#     show_message: bool,
+# }
+#
+# pub enum Msg {
+#     Click(),
+# }
+
+// ...
+#[derive(Properties, Clone, PartialEq)]
+pub struct Props{
+    pub name: String,
+}
+// ...
+
+# impl Component for UseOfPropertyComponent {
+#     type Message = Msg;
+#     type Properties = Props;
+#
+#     fn create(props: Self::Properties, link: ComponentLink<Self>) -> Self {
+#         Self {
+#             link,
+#             props: props.clone(),
+#             name: props.name.into(),
+#             show_message: false,
+#         }
+#     }
+#
+#     fn update(&mut self, msg: Self::Message) -> ShouldRender {
+#         match msg {
+#             Msg::Click() => self.show_message = true,
+#         }
+#         true
+#     }
+#
+#     fn change(&mut self, props: Self::Properties) -> ShouldRender {
+#         if self.props != props {
+#             self.props = props;
+#             true
+#         } else {
+#             false
+#         }
+#     }
+#
+#     fn view(&self) -> Html {
+#         if !self.show_message {
+#             html! {
+#                 <>
+#                     <button onclick=self.link.callback( |_| Msg::Click() )>{"Click here!"} </button>
+#                 </>
+#             }
+#         } else {
+#             html! {
+#                 <>
+#                     <h1>{format!("Hello {}",self.name)}</h1>
+#                 </>
+#             }
+#         }
+#     }
+# }
+
+```
+
+## Attach the property bag in the state
```

Comment:
I think all the titles should be written in the [present participle](https://dictionary.cambridge.org/dictionary/english/present-participle)

Reply:
I am really not a fan.

=> I will do